### PR TITLE
add types and interface for subscriber

### PR DESCRIPTION
**Pull request by mbarekh**

### CODE REVIEW

File: src/communication/xcWebSocketSubscriber.ts [hunk 1]
```diff
@@ -378,26 +367,27 @@ class Subscriber {
...
+    private getJsonDataFromGetXcApiListRequest(data: string): Array<String> {
         let jsonData = this.getJsonData(data);
         return jsonData.Apis;
     };
 
-    private getJsonData(data: string) {
-        return JSON.parse(data.substring(data.indexOf("{"), data.lastIndexOf("}") + 1));
+    private getJsonData(data: string): any {
+        let jsonData = JSON.parse(data.substring(data.indexOf("{"), data.lastIndexOf("}") + 1));
+        return jsonData;
```
Comment (maurelio1234) [inline src/communication/xcWebSocketSubscriber.ts]: Single line?

Comment (mbarekh) [inline src/communication/xcWebSocketSubscriber.ts]: More practical when debugging in two lines

Comment (maurelio1234) [inline src/communication/xcWebSocketSubscriber.ts]: consider join them if you're not debugging this code anymore ;)

Comment (mbarekh) [inline src/communication/xcWebSocketSubscriber.ts]: done

**Pull request merged**